web/source/keymanweb.ts

```diff
@@ -118,12 +118,13 @@ if(!window['keyman']['initialized']) {
     util.attachDOMEvent(document, 'keyup', keymanweb.hotkeyManager._Process, false);
 
     /**
-     * Reset OSK shift states when entering or exiting the active element
+     * Reset context when entering or exiting the active element.
+     * Will also trigger OSK shift state / layer reset.
      **/
     function resetVKShift() {
       let keyman = com.keyman.singleton;
       if(!keyman.uiManager.isActivating && keyman.osk?.vkbd) {
-        keyman.core.keyboardProcessor._UpdateVKShift(null);  //this should be enabled !!!!! TODO
+        keyman.core.resetContext(null);
       }
```

Comment:
Is this change due to my comment from earlier? What's the change in behaviour here?

Reply:
We're not _just_ resetting the OSK here.  We're letting the OSK's reset occur as a side-effect from a context reset... since one should be happening in this scenario, anyway.

As the context reset already has to handle this logic, it centralizes things a bit as well.